### ISSUE TITLE
IAsyncEnumerable not working as expected .Net 7

### ISSUE THREAD

**Issue by AndrewMcAlister** [dotnet/docs#32747]
The below code, when run from browser or swagger, blocks until all data is returned, then displays

        [HttpGet("Numbers")]
        public IAsyncEnumerable<int> Get()
        {
            return FetchItems();
        }

        static async IAsyncEnumerable<int> FetchItems()
        {
            for (int i = 1; i <= 10; i++)
            {
                await Task.Delay(1000);
                yield return i;
            }
        }
    }

I was expecting each int to return when ready
---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 65865d43-7739-9632-2730-3a8b6020a582
* Version Independent ID: fd0f14c8-be87-6ee2-9489-b117cf528647
* Content: [Breaking change: MVC no longer buffers IAsyncEnumerable types when using System.Text.Json - .NET](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/iasyncenumerable-not-buffered-by-mvc)
* Content Source: [docs/core/compatibility/aspnet-core/6.0/iasyncenumerable-not-buffered-by-mvc.md](https://github.com/dotnet/docs/blob/main/docs/core/compatibility/aspnet-core/6.0/iasyncenumerable-not-buffered-by-mvc.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @serpent5
* Microsoft Alias: **gewarren**

**Comment by gewarren** [comment]
@brunolins16 Can you help with this question?

**Comment by neyromant** [comment]
Hi, @brunolins16  I am facing the same problem as @AndrewMcAlister.
This is reproducible when running via dotnet run, but it is not reproducible only if you publish the project and run the executable.

**Comment by fdcastel** [comment]
Also: I didn't find anything mentioned in [Breaking changes in .NET 7](https://learn.microsoft.com/en-us/dotnet/core/compatibility/7.0) which could relate. 

Please advice. This issue is open for 6 months now and it seems a big breaking change.

**Comment by gewarren** [comment]
@Rick-Anderson Could you help to figure out what should be done here?

**Comment by Rick-Anderson** [comment]
@fdcastel  why do you care how it behaves in VS?

**Comment by fdcastel** [comment]
@Rick-Anderson sorry if I misunderstood your question (or I'm missing something here). 

My primary motive for reporting this was because I lost almost one hour trying to understand what I was doing wrong (spoiler: I wasn't). And I believe many other users would have the same problem.

But answering your question: it's also because I love Visual Studio. Is one of the best tools I've ever used in almost 30 years of coding. 

I also use VSCode for Python and TypeScript projects. But for C# and .NET ***nothing*** beats Visual Studio.

_That said, I have to tell you I find the question a bit awkward. Specially coming from someone working for Microsoft 😅_

Kind regards,

Fabio.

**Comment by Rick-Anderson** [comment]
What's the big problem with VS? The code still works, it's only the buffering behavior that surprised you, right?

**Comment by fdcastel** [comment]
Yes. But shouldn't I be surprised? A very basic sample code not working as expected when running from Visual Studio?

I don't know anything about this "buffering behavior" that Visual Studio is showing. I often read the documentation about breaking changes between different versions of the framework and the tools but in this case I found nothing (yet). May you help me?

Is this expected? Where it is documented? How I can disable it?

And why it doesn't happen when I change the project to use .NET 6.0?

**Comment by adegeo** [comment]
_@fdcastel Sorry for what seems like hostility in the thread._

Trying to go through old issues... If there is still an issue or design problem with this, it's better to open it in Visual Studio's feedback system or on the asp.net core source repository.
